src/cluster_legacy.c

```diff
@@ -6532,3 +6851,55 @@ int clusterAllowFailoverCmd(client *c) {
 void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
+
+/* Replicate all migrating and importing slot states to a
+ * a given replica client. */
+void clusterReplicateOpenSlots(void)
+{
+    if (!server.cluster_enabled) return;
+
+    int argc = 5;
+    robj **argv = zmalloc(sizeof(robj*)*argc);
+
+    sds cmd = sdsnew("CLUSTER");
+    sds subcmd = sdsnew("SETSLOT");
+    sds importing = sdsnew("IMPORTING");
+    sds migrating = sdsnew("MIGRATING");
+    robj *cmd_obj = createObject(OBJ_STRING, cmd);
+    robj *subcmd_obj = createObject(OBJ_STRING, subcmd);
+    robj *imp_obj = createObject(OBJ_STRING, importing);
+    robj *mig_obj = createObject(OBJ_STRING, migrating);
+
+    argv[0] = cmd_obj;
+    argv[1] = subcmd_obj;
+
+    for (int i = 0; i < 2; i++) {
+        clusterNode **nodes_ptr = NULL;
+        if (i == 0) {
+            nodes_ptr = server.cluster->importing_slots_from;
+            argv[3] = imp_obj;
+        } else {
+            nodes_ptr = server.cluster->migrating_slots_to;
+            argv[3] = mig_obj;
+        }
+
+        for (int j = 0; j < CLUSTER_SLOTS; j++) {
+            if (nodes_ptr[j] == NULL) continue;
+
+            argv[2] = createStringObjectFromLongLongForValue(j);
+            sds name = sdsnewlen(nodes_ptr[j]->name, sizeof(nodes_ptr[j]->name));
```

Comment:
This feels pretty redundant, shall we loop over each node (primary) and generate the `CLUSTER SETSLOT` command for the slot(s) owned by it.

Reply:
The idea here is to replicate the "migrating/importing" states. If the slot ownership is handled via `clusterUpdateSlotsConfigWith`.

Reply:
We will take a different approach in a follow up PR (#419)